packages/kit/src/runtime/hooks.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @param {...import('types/hooks').SequenceHandle} handles
+ * @returns {import('types/hooks').Handle}
+ */
+export function sequence(...handles) {
+	const length = handles.length;
+	if (!length) return ({ request, resolve }) => resolve(request);
+
+	return ({ request, resolve: super_resolve }) => {
+		return apply_handle(0);
+
+		/**
+		 * @param {number} i
+		 * @returns {import('types/helper').MaybePromise<import('types/hooks').ServerResponse>}
+		 */
+		function apply_handle(i) {
+			const handle = handles[i];
+			const resolve = i >= length - 1 ? super_resolve : () => apply_handle(i + 1);
+			const next = () => resolve(request);
+
+			return handle({ request, resolve, next });
+		}
+	};
+}
```

Comment:
getting rid of `next` (see comment). also, the `resolve` function should take a `request` argument

```suggestion
/**
 * @param {...import('types/hooks').Handle} handlers
 * @returns {import('types/hooks').Handle}
 */
export function sequence(...handlers) {
	const length = handlers.length;
	if (!length) return ({ request, resolve }) => resolve(request);

	return ({ request, resolve }) => {
		return apply_handle(0, request);

		/**
		 * @param {number} i
		 * @param {import('types/hooks').ServerRequest} request
		 * @returns {import('types/helper').MaybePromise<import('types/hooks').ServerResponse>}
		 */
		function apply_handle(i, request) {
			const handle = handlers[i];

			return handle({
				request,
				resolve: i < length - 1 ? (request) => apply_handle(i + 1, request) : resolve
			});
		}
	};
}
```

Reply:
You're totally right, `request` should always be passed to the `resolve` function. The `request` may be mutated or totally cloned by a `Handle`.

I made the change in a new commit.